tests/conftest.py

```diff
@@ -0,0 +1,215 @@
+# don't run overwrite_test_data marked test unless specifically asked to
+_require_explicit_marks = ["overwrite_test_data"]
+
+
+def pytest_collection_modifyitems(config, items):
+    keywordexpr = config.option.keyword
+    markexpr = config.option.markexpr
+    if keywordexpr or markexpr:
+        return  # let pytest handle this
+
+    for item in items:
+        for keyword in _require_explicit_marks:
+            if keyword in item.keywords:
+                skip = pytest.mark.skip(reason=f"{keyword} not explicitly selected")
+                item.add_marker(skip)
+
+
+_grid_kwargs = {
+    GridType.REGULAR: [],
+    GridType.REGULAR_AREA_WEIGHTED: ["area"],
+    GridType.REGULAR_WITH_LAND: ["wet_mask"],
+    GridType.REGULAR_WITH_LAND_AREA_WEIGHTED: ["wet_mask", "area"],
+    GridType.IRREGULAR_WITH_LAND: [
+        "wet_mask",
+        "dxw",
+        "dyw",
+        "dxs",
+        "dys",
+        "area",
+        "kappa_w",
+        "kappa_s",
+    ],
+    GridType.TRIPOLAR_REGULAR_WITH_LAND_AREA_WEIGHTED: ["wet_mask", "area"],
+    GridType.TRIPOLAR_POP_WITH_LAND: ["wet_mask", "dxe", "dye", "dxn", "dyn", "tarea"],
+}
+
+
+scalar_grids = [
+    GridType.REGULAR,
+    GridType.REGULAR_AREA_WEIGHTED,
+    GridType.REGULAR_WITH_LAND,
+    GridType.REGULAR_WITH_LAND_AREA_WEIGHTED,
+    GridType.IRREGULAR_WITH_LAND,
+    GridType.TRIPOLAR_REGULAR_WITH_LAND_AREA_WEIGHTED,
+    GridType.TRIPOLAR_POP_WITH_LAND,
+]
+tripolar_grids = [
+    GridType.TRIPOLAR_REGULAR_WITH_LAND_AREA_WEIGHTED,
+    GridType.TRIPOLAR_POP_WITH_LAND,
+]
+vector_grids = [GridType.VECTOR_C_GRID]
+
+
+def _make_random_data(shape: Tuple[int, int], seed: int) -> np.ndarray:
+    rng = Generator(PCG64(seed))
+    return rng.random(shape)
+
+
+def _make_mask_data(shape: Tuple[int, int]) -> np.ndarray:
+    mask_data = np.ones(shape)
+    ny, nx = shape
+    mask_data[0, :] = 0  #  Antarctica; required for some kernels
+    mask_data[: (ny // 2), : (nx // 2)] = 0
+    return mask_data
+
+
+def _make_irregular_grid_data(shape: Tuple[int, int], seed: int) -> np.ndarray:
+    rng = Generator(PCG64(seed))
+    # avoid large-amplitude variation, ensure positive values, mean of 1
+    grid_data = 0.9 + 0.2 * rng.random(shape)
+    assert np.all(grid_data > 0)
+    return grid_data
+
+
+def _make_irregular_tripole_grid_data(shape: Tuple[int, int], seed: int) -> np.ndarray:
+    rng = Generator(PCG64(seed))
+    # avoid large-amplitude variation, ensure positive values, mean of 1
+    grid_data = 0.9 + 0.2 * rng.random(shape)
+    assert np.all(grid_data > 0)
+    # make northern edge grid data fold onto itself
+    nx = shape[-1]
+    half_northern_edge = grid_data[-1, : (nx // 2)]
+    grid_data[-1, (nx // 2) :] = half_northern_edge[::-1]
+    return grid_data
+
+
+@pytest.fixture(scope="session", params=scalar_grids)
+def scalar_grid_type_data_and_extra_kwargs(request):
+    grid_type = request.param
+    shape = (128, 256)
+
+    data = _make_random_data(shape, 100)
```

Comment:
...and then we invoke the function with a fixed, constant seed, which is the same every time the test is run. That means we always get the same random data.